add history recording and overview

tametsi opened the pull request:
- Add history recording
- Add history page to view & load the last runs

xXjojaXx committed:
Add empty History-Page

xXjojaXx committed:
Add history utility

Implements generic 'History'-class for managing the creation of back-ups in local storage

xXjojaXx committed:
Add requirements for storing in history to  `Run` , `Runner` and `RoundManager`

xXjojaXx committed:
Add content to history page

xXjojaXx committed:
Add design of entries on history page

xXjojaXx committed:
Fix missing reactivity at button clicks

xXjojaXx committed:
Add basic sorting of entries on history page

Newest to the top!

xXjojaXx committed:
Add requirements for storing in history to `Timer`

xXjojaXx committed:
Remove saving of run on changes

xXjojaXx committed:
Revert "Remove saving of run on changes"

This reverts commit c108ca0614ecd61229b65e6fab31ee7b0be19ed2.

xXjojaXx committed:
Recommit "Remove saving of run on changes"

With correct lines selected this time...

xXjojaXx committed:
Refctoring: Save a whole session instead of only the run in history

xXjojaXx committed:
Add confirmation for deletion of history items

xXjojaXx committed:
Add message on History-page if no entries were found

xXjojaXx committed:
Add selection of history sorting-direction and move logic into `History` class

tametsi committed:
Refactor: abstract `HistoryEntry` into own component

tametsi committed:
Update: highlight `HistoryEntry` data & add tooltips for the buttons

tametsi committed:
Add `excludeEmptySessions` setting on history page

tametsi committed:
refactor: map history-entry-sorting-functions by `HistorySorting`

Co-Authored-By: xXjojaXx <73311246+xXjojaXx@users.noreply.github.com>

tametsi committed:
Merge branch 'main' into feat-history

tametsi committed:
refactor: create session-store with `createWriteableObjectStore`

xXjojaXx commented on src/lib/runner.ts:
```diff
@@ -2,14 +2,16 @@ import type Run from '@lib/run';
 		private _alias?: string
-	) {}
+	) {
+		this._rounds = new RoundManager();
```
Switching from inline- to constructor-initialization is not required and should be undone

tametsi replied:
resolved: https://github.com/tametsi/ligas/pull/1/commits/ec83176d8b098346637230f754fca10aa994dab4

xXjojaXx commented on src/lib/runner.ts:
```diff
@@ -48,13 +50,32 @@ export default class Runner {
 class RoundManager {
 	private _rounds: number[] = [];
 
+	constructor() {}
```
Unused constructor should be removed again

tametsi replied:
resolved: https://github.com/tametsi/ligas/pull/1/commits/ec83176d8b098346637230f754fca10aa994dab4

tametsi committed:
refactor: remove unused constructors

- move `_rounds` initialization from constructor to declaration
- remove unused constructor

xXjojaXx reviewed (approved):
Completed
